examples/text/plot_document_clustering.py

```diff
@@ -3,284 +3,423 @@
+df_std = pd.DataFrame(evaluations_std).set_index("estimator")
+
+df.drop(
+    ["train_time"],
+    axis="columns",
+).plot.barh(ax=ax0, xerr=df_std)
+ax0.set_xlabel("Clustering scores")
+ax0.set_ylabel("")
 
-if not opts.use_hashing:
-    print("Top terms per cluster:")
-
-    if opts.n_components:
-        original_space_centroids = svd.inverse_transform(km.cluster_centers_)
-        order_centroids = original_space_centroids.argsort()[:, ::-1]
-    else:
-        order_centroids = km.cluster_centers_.argsort()[:, ::-1]
-
-    terms = vectorizer.get_feature_names_out()
-    for i in range(true_k):
-        print("Cluster %d:" % i, end="")
-        for ind in order_centroids[i, :10]:
-            print(" %s" % terms[ind], end="")
-        print()
+df["train_time"].plot.barh(ax=ax1, xerr=df_std["train_time"])
+ax1.set_xlabel("Clustering time (s)")
+plt.tight_layout()
+
+# %%
+# :class:`~sklearn.cluster.KMeans` and :class:`~sklearn.cluster.MiniBatchKMeans`
+# suffer from the phenomenon called the `Curse of Dimensionality
+# <https://en.wikipedia.org/wiki/Curse_of_dimensionality>`_ for high dimensional
+# datasets such as text data. That is the reason why the overall scores improve
+# when using LSA. Using LSA reduced data also improves the stability and
+# requires lower clustering time, though keep in mind that the LSA step itself
+# takes a long time, especially with hashed vectors.
+#
+# The Silhouette Coefficient is defined between 0 and 1. In all cases we obtain
+# values close to 0 (even if they improve a bit after using LSA) because its
+# definition requires measuring distances, in contrast with other evaluation
+# metrics such as the V-measure and the Adjusted Rand Index which are only based
+# on cluster assignments rather than distances. Notice that strictly speaking,
+# one should not compare the Silhouette Coefficient between spaces of different
+# dimension, due to the different notions of distance they imply.
+#
+# The homogeneity, completeness and hence v-measure metrics do not yield a
+# baseline with regards to random labeling: this means that depending on the
+# number of samples, clusters and ground truth classes, a completely random
+# labeling will not always yield the same values. In particular random labeling
+# won’t yield zero scores, especially when the number of clusters is large. This
+# problem can safely be ignored when the number of samples is more than a
+# thousand and the number of clusters is less than 10, which is the case of the
+# present example. For smaller sample sizes or larger number of clusters it is
+# safer to use an adjusted index such as the Adjusted Rand Index (ARI).
+#
+# The size of the error bars show that :class:`~sklearn.cluster.MiniBatchKMeans`
+# is less stable than :class:`~sklearn.cluster.KMeans` for this relatively small
+# dataset. It is more interesting to use when the number of samples is much
+# bigger, but it comes at the expense of clustering quality.
```

Comment:
```suggestion
# bigger, but it can come at the expense of a small degradation in
# clustering quality compared to the traditional k-means algorithm.
```